prometheus/src/bank_metrics.rs

```diff
@@ -0,0 +1,49 @@
+use solana_runtime::bank::Bank;
+
+use crate::utils::{write_metric, Metric, MetricFamily};
+use std::{io, sync::Arc, time::SystemTime};
+
+pub fn write_bank_metrics<W: io::Write>(
+    at: SystemTime,
+    bank: &Arc<Bank>,
+    out: &mut W,
+) -> io::Result<()> {
+    write_metric(
+        out,
+        &MetricFamily {
+            name: "solana_bank_slot",
+            help: "Current Slot",
+            type_: "gauge",
+            metrics: vec![Metric::new(bank.slot()).at(at)],
+        },
+    )?;
+    write_metric(
+        out,
+        &MetricFamily {
+            name: "solana_bank_epoch",
+            help: "Current Epoch",
+            type_: "gauge",
+            metrics: vec![Metric::new(bank.epoch()).at(at)],
+        },
+    )?;
+    write_metric(
+        out,
+        &MetricFamily {
+            name: "solana_bank_successful_transaction_count",
+            help: "Number of transactions in the block that executed successfully",
```

Comment:
So this is only about the most recent block?

Reply:
yeah, this is the bank at the highest slot

Reply:
Not sure how useful that is, if Prometheus samples every ~15 seconds, it would sample once every ~19 blocks. I guess it could show high-level trends if we average over some time window, but I expect it to be very noisy. A counter would be much nicer, because it doesn’t miss events that happened in between the two sampling events. But if we only sample the bank on demand, then we can’t build a counter, unless Solana already tracks one.

Reply:
humm.. that's fair, I'll delete it for now